topostats/grainstats.py

```diff
@@ -179,39 +175,29 @@ def calculate_stats(self) -> Dict:
 
             if self.save_cropped_grains:
                 output_grain.mkdir(parents=True, exist_ok=True)
-
                 if self.cropped_size == -1:
                     # Plot the cropped grain image
                     plot_and_save(
```

Comment:
I think we might be able to reduce the amount of code here with a simple loop...

```
for name, image in {'processed_grain': grain_image, 'grainmask':grain_mask, 'grain_image': masked_grain)image].items():
    plot_and_save(data=image,
                  output_dir=output_grain,
                  filename=f"{self.image_name}_{name}_{index}"
                  pixel_to_nm_scaling_factor=self.pixel_to_nanometre_scaling,
                  **self.plot_opts[name]
                  )
```

This saves repeating essentially the same code three times.

I've purposefully left out the file extension as, in light of making the format flexible, it may be sensible to add in the file extension to the `self.plot_opts` for each image type.

Reply:
Amazing spot! Added this in now for both the `cropped_size=-1` and `cropped_sizes=<value>`